README.md

```diff
@@ -1,3 +1,199 @@
-The readme is for the Bundle Protocol Core Flight System software distribution.
+# BP - The cFS Bundle Protocol Application
 
+[1. Overview](#1-overview)  
+[2. Build with CMake](#2-build-with-cmake)  
+[3. Command Interface](#3-command-interface)  
+
+----------------------------------------------------------------------
+## 1. Overview
+----------------------------------------------------------------------
+
+The cFS Bundle Protocol Application (bp) is a core Flight System (cFS) application that integrates the Bundle Protocol (BP) library (bplib) with cFS to add RFC9171 Bundle Protocol services to embedded space flight systems.
+
+With bp, cFS can participate in a Delay/Disruption Tolerant Network (DTN) as a DTN Node. cFS bp communicate with other DTN nodes via BP.
+
+----------------------------------------------------------------------
+## 2. Build with CMake
+----------------------------------------------------------------------
+
+#### Prerequisites
+
+1. The build requires the __cmake__ build system and a compiler toolchain (by default __gcc__).
+
+Additionally, the __pkg-config__ tool is used to manage the flags required for dependencies.
+
+The bash statements in Section 2. include the `apt install` statement to install the required toolchain.
+
+3. Create a subdirectory for building a cFS bundle that includes  bp and run CMake to set up the build tree.
+
+Notes:
+- These bash statements mimic the YAML statements in the .github Actions and Workflows scripts. Also, the .github Actions and Workflows specify Ubuntu 22.04 (22.04.4 LTS).
+- Example bash scripts are in [docs/example_scripts](docs/example_scripts)
+- Use of the `$HOME` environment variable is at your discretion. The example scripts use `$HOME` to distinguish the top level folder versus `$HOME/build`, the build folder.
+
+The `install-toolchain` script installs cmake, pkg-config and build-essential (gcc toolchain). Run the script once, if required. There's no harm done by running the script again. It reports that the packages are already installed.
+
+`install-toolchain`
+```sh
+   cd $CFS_HOME
+   sudo apt update
+   sudo apt upgrade
+   sudo apt install cmake pkg-config build-essential
+   git clone https://github.com/intel/tinycbor.git tinycbor-source
+   mkdir tinycbor-staging
+   pushd tinycbor-source
+   make all && make DESTDIR=../tinycbor-staging install
+   popd
+   # Install from tinycbor-staging to /usr/local/include, lib, and bin
+   sudo cp -rv -t / tinycbor-staging/*
+```
```

Comment:
I don't think we should duplicate the example script content into the readme. That'll make it harder to maintain. Instead, perhaps just give the instructions on how to run the example script?

Reply:
Agreed. Working on the change now.